lib/hexpm_web/live/search_live.ex

```diff
@@ -0,0 +1,72 @@
+defmodule HexpmWeb.SearchLive do
+  @moduledoc """
+  A LiveView for live package search from queries.
+  """
+  use HexpmWeb, :live_view
+
+  alias Hexpm.Repository.Packages
+
+  require Logger
+
+  @max_matches 5
+  @matches_sort :recent_downloads
+  @query_size_limit_bytes 100
+
+  @impl true
+  def render(assigns) do
+    HexpmWeb.PageView.render("search.html", assigns)
+  end
+
+  @impl true
+  def mount(_params, _session, socket) do
+    repositories = Enum.map(Hexpm.Accounts.Users.all_organizations(nil), & &1.repository)
+
+    {:ok,
+     assign(socket,
+       container: nil,
+       custom_flash: true,
+       hide_search: true,
+       query: nil,
+       result: nil,
+       loading: false,
+       matches: [],
+       repositories: repositories
+     )}
+  end
+
+  @impl true
+  def handle_event("suggest", %{"search" => query}, socket)
+      when byte_size(query) <= @query_size_limit_bytes do
+    Logger.debug("Got suggest event in #{__MODULE__} with query: #{query}")
+
+    results =
+      Packages.search(
+        socket.assigns.repositories,
+        0,
+        @max_matches,
+        Hexpm.Utils.parse_search(query),
+        @matches_sort,
+        nil
+      )
+
+    {:noreply, assign(socket, matches: results)}
+  end
+
+  @impl true
+  def handle_event("search", %{"search" => query}, socket)
+      when byte_size(query) <= @query_size_limit_bytes do
```

Comment:
With this check it's not possible to do large search queries at all.

Reply:
Should we keep a guard for this? If so, what should be the limit?

Reply:
What should be the limit on 

> This looks good overall but I think the search results should be moved to the dropdown. Now you will get both the browsers suggestions and our search suggestions.
> 
> <img alt="Screenshot 2021-12-08 at 14 56 20" width="704" src="https://user-images.githubusercontent.com/316890/145220465-d460b661-1a76-44c2-8b10-941d4a6575eb.png">

I think the main issue I stumbled into is that the version of Bootstrap in Hex is very old, and does not properly support datalists, so it might need some ugly style hacks.

Reply:
> I think the main issue I stumbled into is that the version of Bootstrap in Hex is very old, and does not properly support datalists, so it might need some ugly style hacks.

I think a dropdown may look better. That way we can also add it to the search in the navigation bar.

Reply:
About the datalist - see comment https://github.com/hexpm/hexpm/pull/1080#issuecomment-990226424